src/main/java/com/google/devtools/build/lib/packages/RuleClass.java

```diff
@@ -241,7 +244,7 @@ public enum ToolchainResolutionMode {
      * href="https://github.com/bazelbuild/bazel/issues/12899#issuecomment-767759147}#12899</a>is
      * addressed, so platforms are unconditionally provided for all rules.
      */
-    HAS_SELECT,
+    NEEDS_TOOLCHAIN_RESOLUTION_FOR_SPECIAL_REASONS,
```

Comment:
I think there's another name which would fit even better here, but a suggestion is escaping me. I'm running off to a meeting but will follow up on this comment.

Overall love your PR and your quick turnaround time on it!

Reply:
What about something like `ONLY_FOR_COMMON_LOGIC`? I'm still not even sure that's the best name. But I'm hoping we could somehow reference that this is for rules where the rule-specific logic doesn't need toolchain resolution, but it'll be optionally available wihen common Bazel logic attached to the rule (like select() and target_compatible_with) needs it.

It's as bit more precise than `SPECIAL_RESONS`, which of course could mean anything.

Reply:
I wasn't sure if you meant literally `ONLY_FOR_COMMON_LOGIC`, or if you meant it as a suffix. Either way, I changed it to `ENABLED_ONLY_FOR_COMMON_LOGIC`.

Reply:
I like this. Thanks.